xcm/procedural/src/v0/multilocation.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use proc_macro2::{Span, TokenStream};
+use quote::{format_ident, quote};
+
+pub fn generate_conversion_functions(input: proc_macro::TokenStream) -> syn::Result<TokenStream> {
+	if !input.is_empty() {
+		return Err(syn::Error::new(Span::call_site(), "No arguments expected"))
+	}
+
+	let from_tuples = generate_conversion_from_tuples();
+	let from_v1 = generate_conversion_from_v1();
+
+	Ok(quote! {
+		#from_tuples
+		#from_v1
+	})
+}
+
+fn generate_conversion_from_tuples() -> TokenStream {
+	let mut from_tuples = TokenStream::new();
+
+	for num_junctions in 0..8usize {
```

Comment:
could this 8 _not_ be hardcoded?

Reply:
yeah, why not an input into the function?